designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,882 @@
+    1. All `.eslintrc` files are ignored.
+    1. `.eslintignore` is ignored.
+    1. `--ignore-file` is ignored.
+    1. `--rulesdir` is ignored.
+    1. `--env` is ignored.
+    1. `--ext` is ignored.
+    1. `eslint-env` config comments are ignored.
+1. If `eslint.config.js` is not found, then fall back to the current behavior.
+
+This keeps the current behavior for the majority of users while allowing some users to test out the new functionality. Also, `-c` could not be used with `eslint.config.js` in this phase.
+
+For every option that is provided and ignored, ESLint will emit a warning. (For example, if `.eslintignore` is found and not used then a warning will be output.)
+
+In the second phase (and in a major release), ESLint will emit deprecation warnings whenever the original functionality is used but will still honor them so long as `eslint.config.js` is not found.
+
+In the third phase (and in another major release), `eslint.config.js` becomes the official way to configure ESLint. If no `eslint.config.js` file is found, ESLint will still search for a `.eslintrc` file, and if found, print an error message information the user that the configuration file format has changed.
+
+So while this is intended to be a breaking change, it will be introduced over the course of three major releases in order to give users ample time to transition.
+
+## Alternatives
+
+While there are no alternatives that cover all of the functionality in this RFC, there are alternatives designed to address various parts.
+
+* Both https://github.com/eslint/rfcs/pull/7 and https://github.com/eslint/rfcs/pull/5 specify an alternative method for resolving plugin locations in configs. This attempts to solve the problem with bundling plugins with configs (https://github.com/eslint/eslint/issues/3458). These proposals have the benefit of working within the current configuration system and requiring very few changes from users.
+* It is possible to come up with a solution to https://github.com/eslint/eslint/issues/8813 that makes use of `extends`, though this would get a bit complicated if an extended config in an `overrides` section also has `overrides`.
+* We could switch the current configuration system over so that all config files are considered to implicitly have `root:true`. This could dramatically simplify configuration searching and merging.
+
+## Open Questions
+
+1. Do we need a command line flag to opt-in to `eslint.config.js` instead of trying to do it alongside the existing configuration system?
+1. Should `eslint` run without any directories, globs, or filenames, fall back to using the globs in the config to find files to run?
+
+## Frequently Asked Questions
+
+### Can a config be an async function?
+
+No. Right now it won't be possible to implement a config with an async function because the rest of ESLint is fully synchronous. Once we look at how to make ESLint more asynchronous, we can revisit and allow configs to be created with async functions.
```

Comment:
Probably can add a note that that #40 is merged so it could eventually be possible when the new format land as stable.

Reply:
True. I'm still planning on doing this as a later step, but there won't be anything blocking it now.